Training/Training/intlist.c

```diff
@@ -0,0 +1,221 @@
+/// <param name="value"></param>
+/// <returns>Error code (if any)</returns>
+int Remove (LinkedList* list, dataType data) {
+	//BaseCase: checks if the list is deleted
+	if (list->isDeleted == true) return E_LIST_DOES_NOT_EXIST;
+	//case1: Checks if the list is empty
+	Node* temp = list->head;
+	Node* prev = NULL;
+	if (temp == NULL) return E_EMPTY_LIST;
+	//case2: Check if the data in head(1st node) is to be removed
+	if (temp != NULL && temp->data == data) {
+		list->head = temp->next;
+		free (temp);
+		return 0;
+	}
+	//case3: if the list is not empty, it iterates to remove the data at spec. index
+	while (temp != NULL && temp->data != data) {
+		prev = temp;
+		temp = temp->next;
+	}
+	//case4: checks if index exceeds list count
+	if (temp != NULL) {
+		prev->next = temp->next;
+		free (temp);
+		return 0;
+	} else {
+		return E_VALUE_NOT_FOUND;
+	}
+}
+
+
+/// <summary>Removes the value at a specified index</summary>
+/// <param name="linkedlist pointer"></param>
+/// <param name="index"></param>
+/// <returns>Error code (if any)</returns>
+int RemoveAt (LinkedList* list, int index) {
+	//BaseCase: checks if the list is deleted
+	if (list->isDeleted == true) return E_LIST_DOES_NOT_EXIST;
+	//case1: Checks if the list is empty
+	Node* temp = list->head;
+	Node* prev = NULL;
+	if (temp == NULL) return E_EMPTY_LIST;
+	//case2: Check if the data in head(1st node) is to be removed
+	if (index == 0) {
+		list->head = temp->next;
+		free (temp);
+		return 0;
+	}
+	//case3: if the list is not empty, it iterates to remove the data at spec. index
+	for (int i = 0; temp != NULL && i < index; i++) {
+		prev = temp;
+		temp = temp->next;
+	}
+	//case4: checks if index exceeds list count
+	if (temp != NULL && prev != NULL) {
+		prev->next = temp->next;
+		free (temp);
+		return 0;
+	} else {
+		return E_INDEX_OUT_OF_RANGE;
+	}
+}
+
+
+/// <summary>Returns the value at a specified index</summary>
+/// <param name="linkedlist pointer"></param>
+/// <param name="index"></param>
+/// <returns>Error code (if any)</returns>
+int Get (LinkedList* list, int index) {
+	//BaseCase: checks if the list is deleted
+	if (list->isDeleted == true) return E_LIST_DOES_NOT_EXIST;
+	//case1: Checks if the list is empty
+	Node* temp = list->head;
+	if (temp == NULL) return E_EMPTY_LIST;
+	//case2: iterates to get the value at index
+	int count = 0;
+	while (temp != NULL) {
+		if (count == index) return (temp->data);
+		count++;
+		temp = temp->next;
+	}
+	return E_INDEX_OUT_OF_RANGE;
+};
+
+
+/// <summary>Inserts a value at a specified index</summary>
+/// <param name="linkedlist pointer"></param>
+/// <param name="value"></param>
+/// <param name="index"></param>
+/// <returns>Error code (if any)</returns>
+int Insert (LinkedList* list, dataType data, int index) {
+	//BaseCase: checks if the list is deleted
+	if (list->isDeleted == true) return E_LIST_DOES_NOT_EXIST;
+	//Case1: Checks if the inserted node is going to be the head
+	Node* newNode = CreateNode (data);

```

Comment:
Strictly speaking you should check whether or not CreateNode returns a NULL. This is a good practice, even if the chance of CreateNode failing is  very small.